Building with clang+ccache fails with -Wself-assign

On master, with clang 703 (OSX):

```
[ 97%] Building CXX object bindings/swig/CMakeFiles/_ikPython.dir/ikPYTHON_wrap.cxx.o
/Users/rdeits/locomotion/drake-distro/drake/pod-build/bindings/swig/ikPYTHON_wrap.cxx:2427:23: error: explicitly assigning value of variable of type 'int' to itself [-Werror,-Wself-assign]
                  res = (res);
                  ~~~ ^  ~~~
/Users/rdeits/locomotion/drake-distro/drake/pod-build/bindings/swig/ikPYTHON_wrap.cxx:2430:23: error: explicitly assigning value of variable of type 'int' to itself [-Werror,-Wself-assign]
                  res = (res);
                  ~~~ ^  ~~~
2 errors generated.
make[6]: *** [bindings/swig/CMakeFiles/_ikPython.dir/ikPYTHON_wrap.cxx.o] Error 1
make[5]: *** [bindings/swig/CMakeFiles/_ikPython.dir/all] Error 2
make[4]: *** [all] Error 2
```

This causes the build to fail because we have `-Werror` set. We can squash the warning with `-Wno-self-assign`, but there's probably a better way.

--- a/drake/doc/fedora.rst
+++ b/drake/doc/fedora.rst
@@ -16,7 +16,7 @@ Install the prerequisites::
     wget -q https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
     sudo rpm -U epel-release-latest-7.noarch.rpm
 
-    sudo yum install autoconf automake bison ccache clang cmake doxygen flex \
+    sudo yum install autoconf automake bison clang cmake doxygen flex \
       freeglut-devel gcc-c++ gcc-gfortran git graphviz gtk2-devel \
       java-*-openjdk-devel libjpeg-turbo-devel libpng-devel libtool make \
       mpfr-devel mpich numpy perl perl-HTML-Form perl-libwww-perl \

--- a/drake/doc/from_source.rst
+++ b/drake/doc/from_source.rst
@@ -137,3 +137,4 @@ To work on the bleeding edge, do::
 	make
 
 This is especially useful if you are ready to contribute your work back to the main repository with a `pull request <https://help.github.com/articles/using-pull-requests/>`_.
+

--- a/drake/doc/homebrew.rst
+++ b/drake/doc/homebrew.rst
@@ -15,7 +15,7 @@ Install the prerequisites::
     brew tap homebrew/science
     brew update
     brew upgrade
-    brew install autoconf automake ccache cmake doxygen gcc glib graphviz gtk+ \
+    brew install autoconf automake cmake doxygen gcc glib graphviz gtk+ \
       jpeg libpng libtool mpfr mpich2 numpy python qt qwt swig valgrind wget
     brew install vtk5 --with-qt
     pip install -U beautifulsoup4 cpplint html5lib Sphinx

--- a/drake/doc/ubuntu.rst
+++ b/drake/doc/ubuntu.rst
@@ -88,7 +88,7 @@ Other prerequisites may be installed as follows::
 
     sudo apt-get update
     sudo apt-get install --no-install-recommends autoconf automake bison \
-      ccache default-jdk doxygen flex freeglut3-dev git \
+      default-jdk doxygen flex freeglut3-dev git \
       graphviz libgtk2.0-dev libhtml-form-perl libjpeg-dev libmpfr-dev \
       libwww-perl libpng-dev libqt4-dev libqt4-opengl-dev libqwt-dev \
       libterm-readkey-perl libtool libvtk-java libvtk5-dev libvtk5-qt4-dev \
@@ -158,15 +158,6 @@ Update the symbolic link in MATLAB to point to the version that was installed ea
     cd /usr/local/MATLAB/R2016a/sys/os/glnxa64
     sudo rm libstdc++.so.6
     sudo ln -s /usr/lib/gcc/x86_64-linux-gnu/4.9/libstdc++.so libstdc++.so.6
-
-ccache
-======
-
-A program called ``ccache`` can be used to speed up (re)builds.
-To use this program, add ``/usr/lib/ccache`` to the front of the ``$PATH``
-environment variable::
-
-    export PATH=/usr/lib/ccache:$PATH
 
 Return to Generic Instructions
 ==============================

